### ISSUE TITLE
Change cron run of main to workflow dispatch event

### ISSUE THREAD

**Issue by stevesloka** [projectcontour/contour-operator#243]
Right now there's a cron schedule which runs the operator's tests each hour to determine if there's an issue from changes in Contour. I was looking into setting up some notifications but if we add a slack or email notification, then we'll get constant alerts. 

We should be able to kick a build of the operator any time something is merged in Contour by utilizing a [workflow dispatch event](https://docs.github.com/en/rest/reference/actions#create-a-workflow-dispatch-event) to trigger the build in the operator automatically and then only get a single resulting alert.

**Comment by danehans** [comment]
@stevesloka as mentioned during [today's community](https://hackmd.io/84Xbl4WBTpm7OBhaOAsSiw) meeting, I think it's a good idea to keep the scheduled runs. This issue should be for **adding** an action to run the operator CI jobs whenever a PR is merged in projectcontour/contour. cc: @jonasrosland

**Comment by stevesloka** [comment]
Yup sounds good @danehans! Maybe we can add the notifications for failures on the pushes to `Contour`, but leave the hourly as-is.
